language-support/scala/codegen/src/main/scala/com/digitalasset/codegen/Main.scala

```diff
@@ -4,47 +4,54 @@
 package com.digitalasset.codegen
 
 import java.io.File
+import java.nio.file.Path
+
+import ch.qos.logback.classic.Level
+import com.digitalasset.daml.lf.codegen.conf.Conf
+import com.typesafe.scalalogging.StrictLogging
+import org.slf4j.{Logger, LoggerFactory}
+
+object Main extends StrictLogging {
+
+  private val codegenId = "Scala Codegen"
+
+  def main(args: Array[String]): Unit =
+    Conf.parse(args) match {
+      case Some(Conf(darMap, outputDir, decoderPkgAndClass, verbosity)) =>
+        setGlobalLogLevel(verbosity)
+        logUnsupportedEventDecoderOverride(decoderPkgAndClass)
+        val (dars, packageName) = darsAndOnePackageName(darMap)
+        CodeGen.generateCode(dars, packageName, outputDir.toFile, CodeGen.Novel)
+      case None =>
+        throw new IllegalArgumentException(
+          s"Invalid $codegenId command line arguments: ${args.mkString(" ")}")
+    }
 
-import scopt.OptionParser
-
-object Main {
-
-  case class Config(
-      inputFiles: Seq[File] = Seq(),
-      packageName: String = "",
-      outputDir: File = new File("."),
-      codeGenMode: CodeGen.Mode = CodeGen.Novel)
-
-  private val parser = new OptionParser[Config]("codegen") {
-    help("help").text("prints this usage text")
-
-    opt[Seq[File]]("input-files").required
-      .abbr("i")
-      .action((d, c) => c.copy(inputFiles = d))
-      .text("input DAR or DALF files")
-
-    opt[String]("package-name")
-      .required()
-      .abbr("p")
-      .action((d, c) => c.copy(packageName = d))
-      .text("package name e.g. com.digitalasset.mypackage")
-
-    opt[File]("output-dir")
-      .required()
-      .abbr("o")
-      .action((d, c) => c.copy(outputDir = d))
-      .text("output directory for Scala files")
+  private def setGlobalLogLevel(verbosity: Level): Unit = {
+    LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME) match {
+      case a: ch.qos.logback.classic.Logger =>
+        a.setLevel(verbosity)
+        logger.info(s"$codegenId verbosity: $verbosity")
+      case _ =>
+        logger.warn(s"$codegenId cannot set requested verbosity: $verbosity")
+    }
   }
 
-  def main(args: Array[String]): Unit = {
-    parser.parse(args, Config()) match {
-      case Some(config) =>
-        CodeGen.generateCode(
-          config.inputFiles.toList,
-          config.packageName,
-          config.outputDir,
-          config.codeGenMode)
-      case None => // arguments are bad, error message will have been displayed
+  private def logUnsupportedEventDecoderOverride(mapping: Option[(String, String)]): Unit =
+    mapping.foreach { decoderMapping =>
+      logger.warn(s"$codegenId does not allow overriding Event Decoder, requested: $decoderMapping")
```

Comment:
```suggestion
      logger.warn(s"$codegenId does not allow overriding Event Decoder, requested: ${decoderMapping: String}")
```

Which will probably not type-check, suggesting a different formatting of the message...?